examples/07_ken_embeddings_example.py

```diff
@@ -0,0 +1,287 @@
+fa2 = FeatureAugmenter(tables=[(embedding_publisher, "Entity")], main_key="Publisher")
+
+X_full = fa1.fit_transform(X)
+X_full = fa2.fit_transform(X_full)
+
+###############################################################################
+# Prediction with regular features
+# --------------------------------
+#
+# We will forget for now the KEN Embeddings and build a typical learning
+# |Pipeline|, where will we try to predict the amount of sales only using
+# the regular features contained in the initial table.
+
+###############################################################################
+# We first use scikit-learn's |ColumnTransformer| to define the columns
+# that will be included in the learning process and the appropriate encoding of
+# categorical variables using the |MinHash| and |OneHotEncoder|:
+from sklearn.compose import make_column_transformer
+
+from sklearn.preprocessing import OneHotEncoder
+from dirty_cat import MinHashEncoder
+
+min_hash = MinHashEncoder(n_components=100)
+ohe = OneHotEncoder(handle_unknown="ignore", sparse_output=False)
+
+encoder = make_column_transformer(
+    ("passthrough", ["Year"]),
+    (ohe, ["Genre"]),
+    (min_hash, ["Platform"]),
+    remainder="drop",
+)
+
+###############################################################################
+# We incorporate our |ColumnTransformer| into a |Pipeline|.
+# We define a predictor, |HGBR|, fast and reliable for big datasets.
+from sklearn.ensemble import HistGradientBoostingRegressor
+from sklearn.pipeline import make_pipeline
+
+hgb = HistGradientBoostingRegressor(random_state=0)
+pipeline = make_pipeline(encoder, hgb)
+
+###############################################################################
+# The |Pipeline| can now be readily applied to the dataframe for prediction:
+from sklearn.model_selection import cross_validate
+
+# We will save the results in a dictionnary:
+all_r2_scores = dict()
+all_rmse_scores = dict()
+
+cv_results = cross_validate(
+    pipeline, X_full, y, scoring=["r2", "neg_root_mean_squared_error"]
+)
+
+all_r2_scores["Regular"] = cv_results["test_r2"]
+all_rmse_scores["Regular"] = -cv_results["test_neg_root_mean_squared_error"]
+
+print("With regular features:")
+print(
+    f"Mean R2 is {all_r2_scores['Regular'].mean():.2f} +-"
+    f" {all_r2_scores['Regular'].std():.2f} and the RMSE is"
+    f" {all_rmse_scores['Regular'].mean():.2f} +-"
+    f" {all_rmse_scores['Regular'].std():.2f}"
+)
+
+###############################################################################
+# Prediction with KEN Embeddings
+# ------------------------------
+#
+# We will now build the learning |Pipeline| using only the KEN embeddings
+# from Wikipedia.
+
+###############################################################################
+# We keep only the embeddings columns:
+encoder2 = make_column_transformer(
+    ("passthrough", emb_columns), ("passthrough", emb_columns2), remainder="drop"
```

Comment:
Is it important to do that?

Reply:
Maybe not, but I wanted to show explicitly what goes into the pipeline. When comparing the three instances you may well see what is included in each.

Reply:
Let's leave it here: I really like the final figure, which really tells a story.